src/main/logutils.ts

```diff
@@ -110,48 +165,153 @@ const tailFile = (path: string) => {
     tailHandler.on("line", function(data: string) {
         handleLogLine(data);
     });
-    
+
     tailHandler.on("error", function(error: unknown ) {
       console.log('ERROR: ', error);
     });
-}    
+}
 
 /**
+ * Splits a WoW combat line intelligently with respect to quotes,
+ * lists, tuples, and what have we.
+ *
+ */
+const splitLogLine = (line: string, maxSplits?: number): LogLine => {
+    const line_len = line.length
+    const list_items: string[][] = [];
+    const args_list: any[] = [];
+    let in_quote = false;
+    let open_lists = 0;
+    let value: any = '';
+
+    // Combat log line always has '<timestamp>  <line>' format,
+    // that is, two spaces between ts and line.
+    const tsIndexEnd = line.indexOf('  ');
+    const timestamp = line.substring(0, tsIndexEnd);
+
+    for (let ptr = tsIndexEnd + 2; ptr < line_len; ptr++) {
+        const c = line.charAt(ptr);
+        if (c === '\n') {
+            break;
+        }
+
+        if (in_quote) {
+            if (c === '"') {
+                in_quote = false;
+                continue;
+            }
+
+        } else {
+            switch (c) {
+            case ',':
+                if (open_lists > 0) {
+                    list_items.at(-1)?.push(value);
```

Comment:
What is the purpose of the `?` here? Null handling? Is that a plausible scenario

Reply:
Yes, exactly that. It's called _optional chaining (?.)_  (documented here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and is used because `.at()` can return `any`, like a string or number. Possible? Yes. Plausible? Not really. I just like to cover my bases since it would avoid an exception of calling `.push()` on a string, which would fail hard and can't be caught, since it'd be a runtime exception - If I recall correctly.